keras/applications/convnext.py

```diff
@@ -0,0 +1,660 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+# pylint: disable=invalid-name
+# pylint: disable=missing-docstring
+# pylint: disable=g-classes-have-attributes
+# pylint: disable=g-direct-tensorflow-import
+"""ConvNeXt models for Keras.
+
+References:
+
+- [A ConvNet for the 2020s](https://arxiv.org/abs/2201.03545)
+  (CVPR 2022)
+"""
+
+from keras import backend
+from keras import layers
+from keras import utils
+from keras import Model
+from keras.applications import imagenet_utils
+from keras.engine import training
+
+import tensorflow.compat.v2 as tf
+from tensorflow.python.util.tf_export import keras_export
+
+BASE_WEIGHTS_PATH = "https://storage.googleapis.com/convnext-tf/keras-applications/convnext/"
+
+WEIGHTS_HASHES = {
+  "tiny":
+    ("dec324e40ebe943afc7b75b72484646eeb092c04bb079df35911d7080364f9a8",
+      "4d4f0e079db2cc0e627b55f7d0d76c367145d14f2c90674415373457cd822346"),
+  "small":
+    ("f964ea5cd5618a1e64902a74ca5ccff3797a4fa5dba11a14f2c4d1a562b72f08",
+      "fd8f0ac74faa4e364d7cb5b2d32af9ae35b54ce5e80525b5beb7b7571320065a"),
+  "base":
+    ("d30e0c509f4e1abe2784d33765d4391ce8fbff259b0bd79f4a63684b20db87d2",
+      "736f7a96cd933ee568611e29f334737fb9aebaaea021ea7adfe4d2f5cbb4a9aa"),
+  "large":
+    ("8a304c66deb782b0d59837bc13127068901adaaa280cfac604d3341aaf44b2cf",
+      "b02b623b3c28586423e6be4aa214e2f5619280b97b4ef6b35ffb686e83235f01"),
+  "xlarge":
+    ("da65d1294d386c71aebd81bc2520b8d42f7f60eee4414806c60730cd63eb15cb",
+      "2bfbf5f0c2b3f004f1c32e9a76661e11a9ac49014ed2a68a49ecd0cd6c88d377"),
+}
```

Comment:
I have also converted the ImageNet-21 checkpoints which would likely be better for transfer learning than the checkpoints from ImageNet-1k pre-training. 

But to add those checkpoints we need the following:

* ImageNet-21k models are supposed to be multi-label classifiers. So the activation should be "sigmoid". So when `weights="imagenet21k" && include_top=True`, `classifier_activation` is supposed to be `sigmoid`.

This would require changes to `imagenet_utils.validate_activation`. As I understand it, it only supports `softmax` at the moment.

Reply:
IMO, a better idea would be to keep the PR as it is. Once it's done, we could work on another PR setting up validation for `sigmoid` when loading pre-trained models like the one mentioned above. After that, I'm happy to work on another PR to incorporate the ImageNet-21k checkpoints making the changes necessary.

Reply:
Sounds good, it's fine to only have support for imagenet1k for now (this is consistent with the other applications). We can add more checkpoints in the future.